corehq/sql_db/models.py

```diff
@@ -0,0 +1,73 @@
+from corehq.util.exceptions import AccessRestricted
+from django.db import models
+
+
+def raise_access_restricted():
+    raise AccessRestricted("This model can be partitioned. Please specify which"
+        "partitioned database with `using` or look for an appropriate dbaccessors "
+        "method.")
+
+
+class RequireDBQuerySet(object):
+    """
+    Takes a QuerySet when instantiated. If .using() is called on this object,
+    it will return the queryset with the given database selected. If anything
+    else is called on this object, it will raise AccessRestricted.
+    """
+
+    def __init__(self, queryset):
+        self.queryset = queryset
+
+    def using(self, db_name):
+        return self.queryset.using(db_name)
+
+    def __getattr__(self, item):
+        raise_access_restricted()
+
+    def __iter__(self):
+        raise_access_restricted()
+
+    def __len__(self):
+        raise_access_restricted()
+
+    def __getitem__(self, key):
+        raise_access_restricted()
+
+
+class RequireDBManager(models.Manager):
+    """
+    A Model Manager which requires that .using() is called before any
+    other methods like .filter() or .get().
+    """
+
+    def get_queryset(self):
+        queryset = super(RequireDBManager, self).get_queryset()
+        return RequireDBQuerySet(queryset)
+
+
+class PartitionedModel(models.Model):
```

Comment:
this purpose of this is to just make it so that we don't accidentally run queries on the wrong db - it essentially forces you to specify a database when calling save(), delete(), or any orm queries against the objects Manager (@snopoke  fyi)